rocketmq-spring-boot/src/main/java/org/apache/rocketmq/spring/config/RocketMQAutoConfiguration.java

```diff
@@ -18,30 +18,37 @@
 package org.apache.rocketmq.spring.config;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.rocketmq.client.MQAdmin;
 import org.apache.rocketmq.client.producer.DefaultMQProducer;
 import org.apache.rocketmq.spring.core.RocketMQTemplate;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
 import org.springframework.context.annotation.Role;
 import org.springframework.util.Assert;
 
-import java.util.Objects;
-
 @Configuration
 @EnableConfigurationProperties(RocketMQProperties.class)
+@ConditionalOnClass({ MQAdmin.class, ObjectMapper.class })
 @ConditionalOnProperty(prefix = "spring.rocketmq", value = "name-server")
 @Import(ListenerContainerConfiguration.class)
+@AutoConfigureAfter(JacksonAutoConfiguration.class)
 public class RocketMQAutoConfiguration {
 
+    @Bean
+    @ConditionalOnMissingBean(ObjectMapper.class)
```

Comment:
should be @ConditionalOnMissingBean(name = "rocketMQMessageObjectMapper")
due to a potential cycle dependency issue. see https://github.com/apache/rocketmq-spring/pull/6#issuecomment-447716369

Reply:
I think it vastly depends what you want to do as part of the starter and it it still unclear to me. If you want to use the "object mapper" defined in the context, then relying on a named bean by default is the wrong thing to do (as you'd force user to name their unique instance that way).

If you want to make sure that a dedicated `ObjectMapper` is used for rocketmq, then it should be more clear than just the name. 

The suggestion here is the best compromise: if you have a single instance we'll use that. If you have more than one, we'll look for the `@Primary` one and if that does not exist we'll fallback to one named `rocketMQMessageObjectMapper`.

Reply:
@walking98 I am also confused why we must create a new ObjectMapper here. I also found too many dedicated named instance in older code. I think we should re-think the reusable components in spring components. We really allow many groupId and many topic existing in one instance, to mapping spring concept, how many instance should we allow?

Reply:
The ideal situation is that there is only exactly one ObjectMapper instance, auto-config needs to create the "rocketMQMessageObjectMapper" if user app does not create. we can not prevent user app to create more than one, if so we just use @primary one instead of creating the "rocketMQMessageObjectMapper".